app/controllers/users_controller.rb

```diff
@@ -87,45 +87,22 @@ def update
     end
   end
 
-  # POST
-  def ban_user
-    if params.has_key?(:duration) && params.has_key?(:site_ban)
-      site_ban = SiteBan.find_by("'#{Time.now.utc}' < expiration AND user_id = #{@user.id}")
-      unless site_ban.nil?
-        site_ban.expiration = Time.now.utc.to_date + params[:duration].to_i.days
-        site_ban.reason = params[:site_ban][:reason]
-        site_ban.save
-      else
-        SiteBan.create(user_id: @user.id, 
-                       expiration: Time.now.utc.to_date + params[:duration].to_i.days, 
-                       reason: params[:site_ban][:reason])
+  def mod_action
+    if params.has_key?(:reason) && params[:reason].present?
+      if params.has_key?(:approve)
+        @user.approve(params[:reason], current_user)
+      elsif params.has_key?(:lift_ban)
+        @user.lift_ban(params[:reason], current_user)
+      elsif params.has_key?(:ban)
+        @user.ban_user(params[:ban].to_i, params[:reason], current_user)
+      elsif params.has_key?(:mark_for_death)
+        @user.mark_for_death(params[:reason], current_user)
       end
     end
```

Comment:
I appreciate that you chose to combine this into one endpoint. URLs should be nouns, not verbs. Good choice.